File: eApartman/eApartman.Model/Korisnik.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel.DataAnnotations;
using eApartman.Model.DataAttributes;
using System.Linq;

namespace eApartman.Model
{
    public class Korisnik
    {
        public int KorisnikId { get; set; }
        [Required(AllowEmptyStrings = false)]
        [MinLength(2)]
        public string Ime { get; set; }
        [Required(AllowEmptyStrings = false)]
        [MinLength(2)]
        public string Prezime { get; set; }
        [MinLength(3)]
        [Required(AllowEmptyStrings = false, ErrorMessage = "Neispravan unos!")]
        public string Username { get; set; }
        [EmailAddress]
        public string Email { get; set; }
        public string Telefon { get; set; }
        public DateTime? DatumRegistracije { get; set; }
        public bool? Status { get; set; }
        public byte[] Slika { get; set; }

        public virtual ICollection<Apartman> Apartmen { get; set; }
        public virtual ICollection<KorisnikUloga> KorisnikUlogas { get; set; }
        public virtual Moderator ModeratorModeratorNavigation { get; set; }
        public int? VlasnikId => ModeratorModeratorNavigation?.VlasnikId;
        public bool? IsVlasnik => KorisnikUlogas?.Any(u => u.UlogaId == 2);
        public bool? IsModeratorApartmani => KorisnikUlogas?.Any(u => u.UlogaId == 4);
        public bool? IsModeratorRezervacije => KorisnikUlogas?.Any(u => u.UlogaId == 5);

        //public virtual ICollection<Rezervacija> Rezervacijas { get; set; }
        //public virtual ICollection<Utisak> Utisaks
        //{
        //    get; set;
        //}
        public int GetVlasnikId()
        {   //Vraća VlasnikId u zavisnosti je li korisnik moderator ili nije
            return IsVlasnik == true ? KorisnikId : int.Parse(VlasnikId.ToString());
        }
        public string GetFullName()
        {
            return $"{Ime} {Prezime}";
        }
    }
}

File: eApartman/UI/eApartman.MobileUI/lib/models/korisnik.dart
import 'dart:convert';

import 'package:eapartman_mobile/models/imodel.dart';
import 'package:eapartman_mobile/models/rezervacija.dart';

class Korisnik implements IModel {
  int korisnikId;
  String username;
  String ime;
  String prezime;
  String email;
  String telefon;
  String password;
  String passwordPotvrda;
  DateTime datumRegistracije;
  List<Rezervacija> rezervacijas;
  List<int> slika;
  List<int> uloge;

  // List<Utisak> utisaks;
  Korisnik({
    this.korisnikId,
    this.username,
    this.ime,
    this.prezime,
    this.email,
    this.telefon,
    this.datumRegistracije,
    this.rezervacijas,
    this.slika,
    this.password,
    this.passwordPotvrda,
    this.uloge,
  });
  factory Korisnik.fromJson(Map<String, dynamic> json) {
    String stringByte = json["slika"] as String;
    List<int> bytes = base64.decode(stringByte);
    return Korisnik(
      korisnikId: json["korisnikId"],
      username: json["username"],
      ime: json["ime"],
      prezime: json["prezime"],
      email: json["email"],
      telefon: json["telefon"],
      datumRegistracije: DateTime.parse(json["datumRegistracije"]),
      rezervacijas: json["rezevacijas"],
      slika: bytes,
    );
  }
  Map<String, dynamic> toJson() => {
        "ime": ime,
        "username": username,
        "prezime": prezime,
        "email": email,
        "telefon": telefon,
        "slika": slika,
        "password": password,
        "passwordPotvrda": passwordPotvrda,
        "uloge": uloge,
      };
}
